### PR TITLE
role implementation

### PR DESCRIPTION
**Pull request by EliosPride**

### CODE REVIEW

File: src/main/java/mate/jdbc/controller/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import mate.jdbc.factory.DriverServiceFactory;
+import mate.jdbc.service.DriverService;
+
+import java.io.IOException;
+
+import static mate.jdbc.util.Constants.DRIVER_ID;
+
+@WebServlet("/admin-controller")
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/controller/AdminController.java]: @EliosPride just "/admin"

File: src/main/java/mate/jdbc/controller/AuthorizationController.java [hunk 2]
```diff
@@ -35,12 +36,16 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
         if (driverOptional.isPresent() && encrypt.equals(driverOptional.get().getPassword())) {
             Driver driver = driverOptional.get();
             session.setAttribute(DRIVER, driver);
-            if (driver.getCarId() == 0) {
-                resp.sendRedirect("/car/add");
+            if (driver.getRole().equals(Role.ADMIN.name())) {
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/controller/AuthorizationController.java]: @EliosPride use Objects.equals()

File: src/main/java/mate/jdbc/controller/RegistrationController.java [hunk 3]
```diff
@@ -40,10 +41,20 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
         String confirmPassword = req.getParameter("confirm");
         String encrypt = HashUtils.encrypt(password);
+        String user = Role.USER.toString();
+        String admin = Role.ADMIN.toString();
         if (password.equals(confirmPassword) && !isEmpty(login, firstName, lastName, password)) {
-            Driver driver = new Driver(firstName, lastName, login, encrypt);
-            driverService.create(driver);
-            resp.sendRedirect("/car/add?" + DRIVER_ID + "=" + driver.getId());
+            if (req.getParameter("button").equals(user)) {
+                Driver driver = new Driver(firstName, lastName, login, encrypt);
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/controller/RegistrationController.java]: @EliosPride pass role to constructor

File: src/main/java/mate/jdbc/controller/RegistrationController.java [hunk 4]
```diff
@@ -40,10 +41,20 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
         String password = req.getParameter("password");
         String confirmPassword = req.getParameter("confirm");
         String encrypt = HashUtils.encrypt(password);
+        String user = Role.USER.toString();
+        String admin = Role.ADMIN.toString();
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/controller/RegistrationController.java]: @EliosPride take role from param "button" and parse to enum

File: src/main/java/mate/jdbc/controller/RegistrationController.java [hunk 5]
```diff
@@ -40,10 +41,20 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
-            resp.sendRedirect("/car/add?" + DRIVER_ID + "=" + driver.getId());
+            if (req.getParameter("button").equals(user)) {
+                Driver driver = new Driver(firstName, lastName, login, encrypt);
+                driver.setRole(user);
+                driverService.create(driver);
+                resp.sendRedirect("/car/add?" + DRIVER_ID + "=" + driver.getId());
+            } else {
+                Driver driver = new Driver(firstName, lastName, login, encrypt);
+                driver.setRole(admin);
+                driverService.create(driver);
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/controller/RegistrationController.java]: @EliosPride extract common logic out of block "if". Use if-else only for redirecting

File: src/main/java/mate/jdbc/filters/AdminFilter.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpSession session = httpServletRequest.getSession();
+        Driver driver = (Driver) session.getAttribute("driver");
+        if (driver.getRole().equals(Role.ADMIN.name())) {
+            chain.doFilter(request, response);
+        } else {
+            ((HttpServletResponse) response).sendRedirect("/");
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/filters/AdminFilter.java]: @EliosPride not redirect to main, user would see 403 forbidden status on the screen

File: src/main/java/mate/jdbc/filters/UserFilter.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest servletRequest = (HttpServletRequest) request;
+        HttpSession session = servletRequest.getSession();
+        Driver driver = (Driver) session.getAttribute("driver");
+        if (driver.getRole().equals(Role.USER.name())) {
+            chain.doFilter(request, response);
+        } else {
+            ((HttpServletResponse) response).sendRedirect("/");
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/filters/UserFilter.java]: @EliosPride the same as for other filter

File: src/main/java/mate/jdbc/model/Driver.java [hunk 8]
```diff
@@ -9,6 +9,15 @@ public class Driver {
     private Long carId;
     private String login;
     private String password;
+    private String role;
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/model/Driver.java]: @EliosPride why string? you must use enum

File: src/main/java/mate/jdbc/controller/AdminController.java (hunk 1 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/AdminController.java]: @Garret-vin +

File: src/main/java/mate/jdbc/controller/AuthorizationController.java (hunk 2 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/AuthorizationController.java]: @Garret-vin done

File: src/main/java/mate/jdbc/controller/RegistrationController.java (hunk 3 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/RegistrationController.java]: @Garret-vin done

File: src/main/java/mate/jdbc/filters/AdminFilter.java (hunk 6 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/filters/AdminFilter.java]: @Garret-vin done

File: src/main/java/mate/jdbc/filters/UserFilter.java (hunk 7 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/filters/UserFilter.java]: @Garret-vin done

File: src/main/java/mate/jdbc/controller/RegistrationController.java (hunk 4 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/RegistrationController.java]: @Garret-vin +

File: src/main/java/mate/jdbc/controller/RegistrationController.java (hunk 5 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/RegistrationController.java]: @Garret-vin +

File: src/main/java/mate/jdbc/model/Driver.java (hunk 8 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/model/Driver.java]: @Garret-vin +

**Pull request merged**